### ISSUE TITLE
Unicode Encode / Decode 

### ISSUE THREAD

**Issue by PseudoCorps** [odoo/odoo#6825]
I think that
- the encode is failing for nav links
- causes blank actions in the nav url links 
    `http://localhost:8069/web#menu_id=4&action=`  
- results in the following error on click or any nav 

```
  File "/Users/user/Documents/Resources/python/Odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 1184, in <genexpr>
    content = ';\n'.join(asset.minify() for asset in self.javascripts)
  File "/Users/user/Documents/Resources/python/Odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 1425, in minify
    return self.with_header(rjsmin(self.content))
  File "/Users/user/Documents/Resources/python/Odoo/odoo/openerp/addons/base/ir/ir_qweb.py", line 1421, in with_header
    return '\n/* %s */\n%s' % (self.name, content)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xef in position 0: ordinal not in range(128)
```

I'm not sure what it was that actually caused the problem, but I the most recent addon I installed that had to do with links was ui_techmenu-8.0.1.8. It just seems like there should be an error handler for blank url parameters (not sure if that is the actual issue). Sorry I don't have too much time to investigate, and I'm about to rebuild from scratch.

Problem: I can't navigate to any pages even old urls, and I can't uninstall the package from the settings page because it results in the same error.

Macbook pro 10.10
Chrome 42 
Odoo 8.0 (master)

![screen shot 2015-05-22 at 12 48 07 pm](https://cloud.githubusercontent.com/assets/2850172/7778541/6510622e-0081-11e5-8e5a-9299a4a7867e.png)

**Comment by PseudoCorps** [comment]
What I said about the empty action thing isn't right because I reloaded and it still has it.

**Comment by xmo-odoo** [comment]
I believe we found that one last week but I do not know if a fix has been validated and when it'll be deployed (/cc @mart-e did you get anything from @blouk or @FalcoBolger on that subject?). The traceback was similar to yours (took us some time to realise this wasn't a one-off as not all machines/configurations have the issue, even with — as far as we know — identical Python and Odoo versions).

The problem is/was that "static" files from theme files are stored as attachments, and during assets collation [attachments are decoded to bytestrings, not unicode](https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/ir/ir_qweb.py#L1407) (whereas [filesystem files are correctly decoded to unicode](https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/ir/ir_qweb.py#L1405)). This becomes an issue if an asset file stored as attachment contains non-ascii characters: because `self.name` is unicode (IIRC), Python implicitly tries to decode `content` using the default `ascii` codec and fails with the traceback you got here.

A possible temporary fix is to find out the problematic CSS file in the attachments and replace non-ascii characters by [unicode escapes](https://mathiasbynens.be/notes/css-escapes).

tl;dr: missing decoding of static files stored as attachments, generally stems from themes.

**Comment by jperville** [comment]
Hello, any chance to get the above fix for this trivial issue in the Odoo 8.0 nightlies?

**Comment by jperville** [comment]
The issue also affects Odoo 9.0 btw, with the same fix.

**Comment by mart-e** [comment]
@jperville could you please give a clear way to reproduce this in 8 and/or 9? I can reproduce when importing a module containing CSS with unicode in base_import module but just want to be sure in case you reproduce it differently.

**Comment by jperville** [comment]
Hi @mart-e , I encountered the issue when investigating a friend's customer odoo "not loading". Firebug pointed me to asset being served with error 500, which I traced to a custom theme with unicode content; then I found about this issue, and about your fix, which solved the problem for the moment.

**Comment by Yenthe666** [comment]
Hi @PseudoCorps ,

Thank you for your bug report.
At this point Odoo 8 is no longer supported as Odoo only supports three versions (9.0, 10.0, 11.0 at the moment).
If you can still reproduce this issue on Odoo 9, Odoo 10 or Odoo 11 we encourage you to create a new bug report.

Regards,
Yenthe
